subsys/net/ip/tcp.c

```diff
@@ -3329,7 +3318,7 @@ static enum net_verdict tcp_in(struct tcp *conn, struct net_pkt *pkt)
 		 * -   & ACK -> TCP_FIN_WAIT_2
 		 * FIN & ACK -> TCP_TIME_WAIT
 		 */
-		if (th) {
+		do {
```

Comment:
Why wrapping this code in a dummy `do { } while (0)` loop? I think we should be consistent and just reduce the indentation everywhere, let's do it once and for good.